tests/e2e/e2e_suite_test.go

```diff
@@ -53,18 +59,76 @@ var _ = BeforeSuite(func() {
 		Expect(errors.New(errString)).NotTo(HaveOccurred())
 	}
 
-	credData, err := utils.ReadFile(cloud)
-	Expect(err).NotTo(HaveOccurred())
-	err = CreateCredentialsSecret(credData, namespace, GetSecretRef(credSecretRef))
-	Expect(err).NotTo(HaveOccurred())
-
 	dpaCR = &DpaCustomResource{
-		Namespace: namespace,
+		Namespace:     namespace,
+		Credentials:   credFile,
+		CredSecretRef: credSecretRef,
+		Provider:      provider,
 	}
 	dpaCR.CustomResource = Dpa
 	testSuiteInstanceName := "ts-" + instanceName
 	dpaCR.Name = testSuiteInstanceName
+	openshift_ci_bool, _ := strconv.ParseBool(openshift_ci)
+	dpaCR.OpenshiftCi = openshift_ci_bool
+
+	if openshift_ci_bool == true {
+		switch dpaCR.Provider {
+		case "aws":
+			cloudCredData, err := utils.ReadFile(dpaCR.Credentials)
```

Comment:
Instead of switching over each provider, can do:
```
cloudCredData, err := utils.ReadFile(dpaCR.Credentials)
Expect(err).NotTo(HaveOccurred())
err = CreateCredentialsSecret(cloudCredData, namespace, fmt.Sprintf("bsl-cloud-credentials-$v", dpaCr.Provider)
Expect(err).NotTo(HaveOccurred())
dpaCR.Credentials = ci_cred_file
```